pages/gallery/gallery.js

```diff
@@ -0,0 +1,420 @@
+      this.dataJSON.tags.forEach(tag => {
+        if (!this.options.tags.includes(tag)) this.options.tags.push(tag)
+      })
+
+      // if no data : no textures to display
+      if (!this.dataJSON.paths) return;
+
+      // starting with the last element of a texture, paths:
+      this.dataJSON.paths.forEach(path => {
+        if (path.versions.includes(this.version === 'latest' ? settings.versions[this.edition][0] : this.version)) {
+          this.displayed.paths[path.id] = {
+            versions: path.versions,
+            path: path.path,
+            useID: path.useID,
+            pathID: path.id
+          }
+        }
+      })
+
+      // re-map array to be objects
+      let tmp = this.dataJSON.uses
+      this.dataJSON.uses = {}
+      tmp.forEach(t => {
+        this.dataJSON.uses[t.id] = {
+          edition: t.editions[0],
+          useID: t.id.toString(),
+          textureID: t.textureID.toString()
+        }
+      })
+
+      tmp = this.dataJSON.textures
+      this.dataJSON.textures = {}
+      tmp.forEach(t => {
+        this.dataJSON.textures[t.id] = {
+          name: t.name,
+          tags: t.type,
+          textureID: t.id
+        }
+      })
+
+      // then we keep uses that are in paths
+      Object.values(this.displayed.paths).forEach(path => {
+        const useID = path.useID
+        const use = this.dataJSON.uses[useID]
+        if (use) this.displayed.uses[useID] = { ...use, useID: useID, pathID: path.pathID }
+      })
+
+      Object.values(this.displayed.uses).forEach(use => {
+        const textureID = use.textureID
+        const texture = this.dataJSON.textures[textureID]
+        if (texture) this.displayed.textures[textureID] = { ...texture, textureID: textureID, useID: use.useID, pathID: use.pathID }
+      })
+
+      this.dataJSON.contributors.forEach(contributor => {
+        this.displayed.contributors[contributor.id] = {
+          username: contributor.username
+        }
+      })
+
+      // stuff below are only available in compliance, not default resolution
+      if (this.current.resolution === '16x') {
+        this.displayedTextures = this.splittedTextures()
+        return
+      }
+
+      Object.values(this.dataJSON.contributions).forEach(contribution => {
+        if (!this.displayed.contributions[contribution.res][contribution.textureID]) this.displayed.contributions[contribution.res][contribution.textureID] = []
+
+        this.displayed.contributions[contribution.res][contribution.textureID].push({
+          contributors: contribution.contributors,
+          date: contribution.date
+        })
+      })
+
+      this.displayedTextures = this.splittedTextures()
+    },
+    scroll() {
+      window.onscroll = () => {
+        let bottomOfWindow = Math.max(window.pageYOffset, document.documentElement.scrollTop, document.body.scrollTop) + window.innerHeight === document.documentElement.offsetHeight
+
+        if (bottomOfWindow) {
+          this.displayedResults += 30
```

Comment:
We can't change language (at the bottom of the page) then? it will always load more results ? Either you put a button to load more results or you change the language option location

Reply:
Then we should change language selection location

Reply:
in the menu with the ``translate`` or ``web`` icon before the banner ?